include/sparrow/dynamic_bitset.hpp

```diff
@@ -62,7 +62,7 @@ namespace sparrow
         size_type size() const noexcept;
         size_type null_count() const noexcept;
 
-        bool test(size_type pos) const;
+        const_reference test(size_type pos) const;
```

Comment:
Could you explain why this change is necessary? To me `test` means, give me the value as of right-now, not give me access to the boolean.

Reply:
`const_reference` is `bool` here. Changing this return type is necessary for `using`-declaration consistency which was required.

Reply:
Ok so I think here `test()` should return `value_type` and, as a side note not related to this pr, `const_reference` should not be `bool` but either an actual reference to bool or a const `bitset_reference`
Anyway, here `test()` really is a capture of the current value, so `value_type` seems logical.

Reply:
Actually yes we should remove revert this changes.

I was mislead by both the present of `test` and of `operator[]`.

Reply:
The confusion is understandable 👍🏽